### ISSUE TITLE
app/tribe_client/tribe_client.js

### ISSUE THREAD

**Issue by dongbohu** [greenelab/adage-server#80]
```
...
    getPromise: function() {
      if (!promise) {
        promise = User.get({}, function(data) {  // line 18
          if (data.meta.total_count !== 0) {
            user = data.objects[0];
          }
        });  // line 22
      }
      return promise;
    },
...
```
On line 18, `User.get(...)` actually returns an instance of Resource class, not a promise. So it seems to make sense to change line 22 into:
```}).$promise;```

All the other places that uses
```UserFactory.getPromise().$promise```
can be simplified into:
```UserFactory.getPromise()```

Reference:
https://docs.angularjs.org/api/ngResource/service/$resource

**Comment by rzelayafavila** [comment]
Ah true, good catch @dongbohu! This makes sense.

**Comment by rzelayafavila** [comment]
Fixed with PR #117
